File: tests/fixtures/java/CryptoIssues.java
// Test fixtures for Java cryptography issues
import java.security.*;
import java.util.Random;

public class CryptoIssues {

    // Should trigger JAVA_CRY001
    public void weakMd5Hash(byte[] data) throws NoSuchAlgorithmException {
        MessageDigest digest = MessageDigest.getInstance("MD5");
        byte[] hash = digest.digest(data);
    }

    // Should trigger JAVA_CRY001
    public void weakSha1Hash(byte[] data) throws NoSuchAlgorithmException {
        MessageDigest digest = MessageDigest.getInstance("SHA-1");
        byte[] hash = digest.digest(data);
    }

    // Should trigger JAVA_CRY002
    public long generateSessionId() {
        Random random = new Random();
        return random.nextLong();
    }

    // Should trigger JAVA_CRY002
    public String generateToken() {
        Random random = new Random();
        return String.valueOf(random.nextInt());
    }
}

File: tests/fixtures/rust/crypto_issues.rs
// Test fixtures for Rust cryptography issues

// Should trigger RUST_CRY001
use md5;
use sha1;

fn test_weak_hashing() {
    use md5::{Md5, Digest};

    let data = b"hello world";
    let mut hasher = Md5::new();
    hasher.update(data);
    let result = hasher.finalize();
}

fn test_sha1_usage() {
    use sha1::{Sha1, Digest};

    let data = b"password123";
    let mut hasher = Sha1::new();
    hasher.update(data);
    let result = hasher.finalize();
}
